### DISCUSSION TITLE
Streaming the result of a stepwise planner?

### DISCUSSION THREAD

**Discussion by cmoski** [microsoft/semantic-kernel#3575] [Q&A] [answered]
I'd like to stream the results of a Stepwise Planner. Is that possible?

Code to illustrate what I mean.. Is it possible to stream the result of the plan to the user in any way?

I realize this could be a dumb question, but I'd like to enable streaming in my Chatbot interface, but also allow for running function calls. I could simulate this of course by cacheing up the results of the planner and streaming them to the user.. 

Obviously doesn't work, but sort of illustrates what I want to be able to do:

```
            var planner = new StepwisePlanner(_kernel, new StepwisePlannerConfig() { MaxTokens = 32000 });
            var plan = planner.CreatePlan(question);
            var result = await _kernel.RunAsync(plan);

            await foreach (var streamingText in _chatGPT.GenerateMessageStreamAsync(_chatHistory))
            {
                Console.WriteLine(streamingText);
            }

```

**Comment by maxupp** [comment]
This is of interest to us too.

**Comment by evchaki** [comment, marked as answer]
@cmoski - check out the new updates for planners - https://devblogs.microsoft.com/semantic-kernel/migrating-from-the-sequential-and-stepwise-planners-to-the-new-handlebars-and-stepwise-planner/
